test/test_utils/sagemaker.py

```diff
@@ -302,8 +282,9 @@ def execute_local_tests(image, pytest_cache_params):
     Run the sagemaker local tests in ec2 instance for the image
     :param image: ECR url
     :param pytest_cache_params: parameters required for :param pytest_cache_util
-    :return: None
+    :return: True if test execution was successful, else False
     """
+    test_success = False
```

Comment:
why do we need to track execution success/failure this way as opposed to just raising exceptions?

Reply:
The problem with python multiprocessing is that it isn't sophisticated enough to handle Exceptions from multiprocessed functions.

The difference between implicitly returning None (by not having a return value in a function) and raising an Exception in the child function, is that the parent function that calls children in multiple processes tries to unpack the Exception returned by the child as "any other return value", and fails to do so, which leads to the weird `cannot pickle '_thread.lock' object` error.

The fix is to always return something. Unfortunately, there is no better solution I could find that would still prevent the silent failures in the SM Local tests.